File: gateway/test/approved-calling-backbone.test.js
const test = require('node:test');
const assert = require('node:assert/strict');

const {
  SERVICE_CONTRACT,
} = require('../src/service-backbones/approved-calling');

test('approved-calling remains a disabled service backbone', () => {
  assert.equal(SERVICE_CONTRACT.lifecycle, 'backbone');
  assert.equal(SERVICE_CONTRACT.enabledByDefault, false);
  assert.equal(SERVICE_CONTRACT.customerVisible, false);
  assert.equal(SERVICE_CONTRACT.minimumPlan, 'essential');
  assert.equal(SERVICE_CONTRACT.callDirection, 'approved-guardian-to-watch-only');
  assert.deepEqual(SERVICE_CONTRACT.protocolCommands, ['PHBX']);
  assert.ok(!SERVICE_CONTRACT.protocolCommands.includes('CALL'));
  assert.ok(SERVICE_CONTRACT.provenBehaviors.includes('unknown number is blocked'));
  assert.ok(SERVICE_CONTRACT.backendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.frontendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.acceptanceGates.length > 0);
  assert.ok(Object.isFrozen(SERVICE_CONTRACT));
});

File: apps/mobile/test/approved_calling_backbone_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:guardian/services/backbones/approved_calling_backbone.dart';
import 'package:guardian/services/guardian_entitlements.dart';

void main() {
  test('approved-calling stays hidden until its acceptance gates pass', () {
    expect(ApprovedCallingBackbone.lifecycle, 'backbone');
    expect(ApprovedCallingBackbone.enabledByDefault, isFalse);
    expect(ApprovedCallingBackbone.customerVisible, isFalse);
    expect(ApprovedCallingBackbone.minimumPlan, 'essential');
    expect(
      ApprovedCallingBackbone.callDirection,
      'approved-guardian-to-watch-only',
    );
    expect(ApprovedCallingBackbone.protocolCommands, <String>['PHBX']);
    expect(ApprovedCallingBackbone.protocolCommands, isNot(contains('CALL')));
    expect(
      ApprovedCallingBackbone.provenBehaviors,
      contains('unknown number is blocked'),
    );
    expect(ApprovedCallingBackbone.frontendMilestones, isNotEmpty);
    expect(ApprovedCallingBackbone.acceptanceGates, isNotEmpty);
    expect(
      GuardianFeature.twoWayCalls.label,
      'Family can call the watch',
    );
  });
}
